README.md

```diff
@@ -22,7 +22,7 @@ guid := "2we34-32ed3-33d33-23rd3"
 counter := resources.ResourceCounter{
     Query: resources.ResourceCounterQuery{
         Namespaces: namespaces,
-        OfType:     &apps.Deployment{},
+        GVK:     CommonGVKs.Deployment,
```

Comment:
I see what you've done here and though I like it, the common way that k8s uses to pass around types for the client, is to pass empty objects. When we you set up watches etc, you pass in empty types that it internally converts to GVKs and uses. While there is an efficiency there, we have used this same paradigm in multiple places through the client facing code, and I think I'd prefer to stick with it, thoughts?

Reply:
Hackathon, been off the grid - coming back to this now.

I think `MakeQuery` does exactly what you're looking for:

```go
query, _ := resources.MakeQuery(&apps.Deployment{}, *scheme, namespaces, o.GetUID())

counter := resources.ResourceCounter{
    Query: query,
    ReadyRequirements: []resources.ResourceConditionReadyRequirements{{
        Type:   "Available",
        Status: "True",
    }},
```

Its been a while since we walked about this, but I think that's what you suggested I implement: a factory method that takes a type specimen and a scheme and then constructs the query for you. I could make the factory method return the ResourceCounter instead of the query, I just felt that was a bit more work than a single method should do, but I can do it.

Reply:
I'm happy with submitting the Query separately, but I'd like to keep the interface of passing in the specimen so that it is aligned with the other tools in the framework.

Reply:
Thanks for the input. Unfortunately, there's a communication / understanding gap here. I am obviously misunderstanding what you are asking of me. Could you provide me an example of what you want this to look like? Then I'll implement it that way. The way I read your input I've already done what you asked, so I'm not sure what to do without an example. Sorry :(